<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Viewer</class>
 <widget class="QMainWindow" name="Viewer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1550</width>
    <height>886</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Viewer</string>
  </property>
  <property name="styleSheet">
   <string>
    QMainWindow {
        background-color: #2c3e50;
    }
    QFrame {
        background-color: #34495e;
        border: 2px solid #1abc9c;
        border-radius: 10px;
    }
    
    QLabel {
        color: #ecf0f1;
        font-size: 14px;
    }
    QDoubleSpinBox {
        background-color: #ecf0f1;
        color: #2c3e50;
        border: 1px solid #1abc9c;
        border-radius: 5px;
    }
    QPushButton {
        background-color: #1abc9c;
        color: #ecf0f1;
        border: none;
        padding: 8px 15px;
        border-radius: 5px;
    }
    QPushButton:hover {
        background-color: #16a085;
    }
    QRadioButton {
        color: #ecf0f1;
    }
    QRadioButton::indicator {
        background-color: #1abc9c;
        border: 1px solid #16a085;
        width: 15px;
        height: 15px;
    }
    QRadioButton::indicator:checked {
        background-color: #16a085;
        border: 1px solid #1abc9c;
    }
    
   </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>371</width>
      <height>571</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="edges_amount">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>480</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Ребра</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_move_z">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>210</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-1000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_move_y">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>169</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Move y:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_scale">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>90</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="decimals">
      <number>2</number>
     </property>
     <property name="minimum">
      <double>0.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
     <property name="value">
      <double>1.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_z">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>330</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">b</string>
     </property>
     <property name="minimum">
      <double>-360.000000000000000</double>
     </property>
     <property name="maximum">
      <double>360.000000000000000</double>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_open_choose_file">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Выбрать файл</string>
     </property>
    </widget>
    <widget class="QLabel" name="vertexes_amount">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>430</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_filename">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>380</y>
       <width>91</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Имя файла</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_z">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>330</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate z</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_y">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>290</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-360.000000000000000</double>
     </property>
     <property name="maximum">
      <double>360.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_vertexes_amount">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>430</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Вершины</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_x">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>249</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate x:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_move_x">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>129</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Move x:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_move_x">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>130</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-1000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_rotate_x">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>250</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-360.000000000000000</double>
     </property>
     <property name="maximum">
      <double>360.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_move_z">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>209</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Move z:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_y">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>289</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate y:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_scale">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>90</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Scale:</string>
     </property>
    </widget>
    <widget class="QLabel" name="filename">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>380</y>
       <width>211</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="doubleSpinBox_move_y">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>170</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-1000.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_clear_changes">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>30</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Сбросить положение</string>
     </property>
    </widget>
    <widget class="QLabel" name="edges_amount_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>480</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
   </widget>
   <widget class="openglView" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>20</y>
      <width>761</width>
      <height>751</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_gif">
    <property name="geometry">
     <rect>
      <x>1400</x>
      <y>240</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>gif</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_dashed">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>70</y>
      <width>70</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>dashed</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_model_color">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>130</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Model color:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_background_color">
    <property name="geometry">
     <rect>
      <x>1350</x>
      <y>130</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Background color:</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_parallel">
    <property name="geometry">
     <rect>
      <x>1290</x>
      <y>40</y>
      <width>73</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>parallel</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_solid">
    <property name="geometry">
     <rect>
      <x>1290</x>
      <y>70</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>solid</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_circle">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>100</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>circle</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_central">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>40</y>
      <width>73</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>central</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_background_color">
    <property name="geometry">
     <rect>
      <x>1490</x>
      <y>130</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	background-color: rgb(222, 221, 218)
	color: black;
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_thickness">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>180</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Thickness:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_jpeg">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>240</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>jpeg</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_bmp">
    <property name="geometry">
     <rect>
      <x>1190</x>
      <y>240</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>bmp</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_figure_color">
    <property name="geometry">
     <rect>
      <x>1290</x>
      <y>130</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton {
	background-color: rgb(53, 132, 228);
	color: black;
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_projection">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>40</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Projection:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_edges">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>70</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Edges:</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_none">
    <property name="geometry">
     <rect>
      <x>1290</x>
      <y>100</y>
      <width>60</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>none</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="doubleSpinBox_thikness">
    <property name="geometry">
     <rect>
      <x>1280</x>
      <y>180</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <double>1.000000000000000</double>
    </property>
    <property name="maximum">
     <double>100.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_vertexes">
    <property name="geometry">
     <rect>
      <x>1181</x>
      <y>101</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertixes:</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_square">
    <property name="geometry">
     <rect>
      <x>1470</x>
      <y>104</y>
      <width>66</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>square</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1550</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>openglView</class>
   <extends>QOpenGLWidget</extends>
   <header>myopenglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
